integration-tests/src/test/java/org/zowe/apiml/integration/graphql/BookControllerTest.java

```diff
@@ -72,13 +79,396 @@ public void whenGetAllBooksWithWrongSchema_thenReturnError() throws SSLException
         }
         """;
 
-        try {
+        AssertionError thrown = assertThrows(AssertionError.class, () -> {
             tester.document(document)
                 .execute()
                 .path("getAllBooks")
-                .entityList(BookConfiguration.class)
-                .hasSizeLessThan(1);
-        } catch (AssertionError ignored) {}
+                .entityList(Book.class)
+                .get();
+        });
+        assertNotNull(thrown);
+    }
+
+    @Test
+    public void whenAddBook_thenReturnAddedBook() throws SSLException {
+        SslContext sslContext = SslContextBuilder
+            .forClient()
+            .trustManager(InsecureTrustManagerFactory.INSTANCE)
+            .build();
+        HttpClient httpClient = HttpClient.create().secure(t -> t.sslContext(sslContext));
+
+        WebTestClient client =
```

Comment:
this code is repeated in every test method, so it can be moved into a @BeforeEach method, which would reduce redundancy.